core/local/atom/await_write_finish.js

```diff
@@ -192,16 +202,28 @@ function debounce(waiting /*: WaitingItem[] */, events /*: AtomEvent[] */) {
               addDebugInfo(event, e)
               // Preserve the action from the first event (it can be a created file)
               event.action = e.action
+
+              if (e.action === 'renamed') {
+                event.oldPath = e.oldPath
+              }
             }
 
-            if (event.action === 'deleted' && e.action === 'created') {
-              // It's just a temporary file that we can ignore
-              log.debug(
-                { createdEvent: e, deletedEvent: event },
-                `Ignore ${e.kind} ${e.action} then ${event.action}`
-              )
-              events.splice(i, 1)
-              i--
+            if (event.action === 'deleted') {
+              if (e.action === 'created') {
+                // It's just a temporary file that we can ignore
+                log.debug(
+                  { createdEvent: e, deletedEvent: event },
+                  `Ignore ${e.kind} ${e.action} then ${event.action}`
+                )
+                events.splice(i, 1)
+                i--
+              } else if (e.action === 'renamed') {
+                addDebugInfo(event, e)
+                // Delete document at oldPath instead of moving then deleting
+                if (e.oldPath) {
```

Comment:
Why do we check that `e` has an `oldPath`? Can we have a `renamed` event without it?

Reply:
It's for Flow. I could make a check method that checks both if you find it clearer.